boilingcore/text_helpers.go

```diff
@@ -38,12 +38,23 @@ import (
 // fk == table = industry.Industry | industry.Industry
 // fk != table = industry.ParentIndustry | industry.Industry
 func txtNameToOne(fk drivers.ForeignKey) (localFn, foreignFn string) {
-	foreignFn = strmangle.Singular(trimSuffixes(fk.Column))
-	fkeyNotTableName := foreignFn != strmangle.Singular(fk.ForeignTable)
-	foreignFn = strmangle.TitleCase(foreignFn)
+	fkColumnTrimmedSuffixes := strmangle.Singular(trimSuffixes(fk.Column))
+	fkeyNotTableName := fkColumnTrimmedSuffixes != strmangle.Singular(fk.ForeignTable)
+	singularForeignTable := strmangle.Singular(fk.ForeignTable)
+
+	if fkColumnTrimmedSuffixes == singularForeignTable {
+		foreignFn = strmangle.TitleCase(strmangle.Singular(fk.Table) + "_" + fkColumnTrimmedSuffixes)
+		if fk.Column != singularForeignTable {
+			foreignFn = strmangle.TitleCase(fkColumnTrimmedSuffixes)
+		}
+	} else if fkColumnTrimmedSuffixes == fk.Column && strings.HasSuffix(fk.Column, "_by") {
```

Comment:
This is not exactly what was spec'd out. A `_by` suffix doesn't mean anything in particular and is just an example, we should not be giving special behaviors to columns with this suffix.

A PR that fixes this correctly would first start with the code that bugzpodder suggested. This should correct the two conflicts in bold in the table. Then some additional code should be added to avoid the `UserUser` name (it should end up as `VideoUser` as discussed below the table).